Makefile

```diff
@@ -673,19 +674,33 @@ doc-html doc-latexpdf doc-epub doc-livehtml : doc-% : doxygen
 
 doc : doc-html
 
-style-c :
-	@astyle --options=.astylerc \
-          $(filter-out include/ceedf.h $(wildcard tests/t*-f.h), \
-            $(wildcard include/*.h interface/*.[ch] tests/*.[ch] backends/*/*.[ch] \
-              examples/*/*/*.[ch] examples/*/*.[ch] examples/*/*.[ch]pp gallery/*/*.[ch]))
+# Style/Format
+CLANG_FORMAT ?= clang-format
+
+FORMAT_OPTS += -style=file -i
+
+%.format : %
+	$(CLANG_FORMAT) $(FORMAT_OPTS) $^
```

Comment:
This is a phony target so it's going to run on every file every time (even if nothing has been edited). Seems to me it may as well be one invocation of `clang-format` rather than one per file.

Reply:
At least on my machine, `make format -j` goes faster with multiple targets, unless I'm missing something.

Reply:
What does this look like for you?
```console
$ time make -j12 format > /dev/null
4.659 real   11.453 user   6.545 sys   386.38 cpu
$ time clang-format -i $(git ls-files *.[ch]pp *.[ch])
2.293 real   2.256 user   0.036 sys   99.99 cpu
```

Reply:
We don't want a hard dependency on GNU parallel, but this is faster. I almost never call this directly so I don't have strong opinions.
```console
$ time ( git ls-files *.[ch]pp *.[ch] | parallel -j8 -N 40 clang-format -i )
0.903 real   3.053 user   0.303 sys   371.55 cpu
```

Reply:
Hmm, I must have mangled something in the attempt